common/test/acceptance/pages/lms/edxnotes.py

```diff
@@ -538,3 +538,46 @@ def text(self, value):
         Sets text for the note.
         """
         self.q(css=self._bounded_selector(".annotator-item textarea")).first.fill(value)
+
+    @property
+    def tags(self):
+        """
+        Returns the tags associated with the note.
+
+        Tags are returned as a list of strings, with each tag as an individual string.
+        """
+        tag_text = []
+        self.show()
+        tags = self.q(css=self._bounded_selector(".annotator-annotation > div.annotator-tags > span.annotator-tag"))
+        if tags:
+            for tag in tags:
+                tag_text.append(tag.text)
+        self.q(css="body").first.click()
+        self.wait_for_notes_invisibility()
+        return tag_text
+
+    @tags.setter
+    def tags(self, tags):
+        """
+        Sets tags for the note. Tags should be supplied as a list of strings, with each tag as an individual string.
+        """
+        self.q(css=self._bounded_selector(".annotator-item input")).first.fill(" ".join(tags))
+
+    def has_sr_label(self, index, expected_text):
+        """
+        Returns true iff a screen reader label exists for the annotator field with the specified index and text.
+        """
+        label_exists = False
+        EmptyPromise(
+            lambda: len(self.q(css=self._bounded_selector("li.annotator-item > label.sr"))) > index,
+            "sr label never appeared"
```

Comment:
Isn't the `EmptyPromise` `description` parameter supposed to describe what is being waited for and not what went wrong?

http://bok-choy.readthedocs.org/en/latest/api_reference.html#bok_choy.promise.EmptyPromise